### PR TITLE
Feature/itunes categories

### LINKED ISSUE #143: itunes:category setzbar machen
<p>Im RSS-Feed fehlt noch der Kategorie-Tag. Diesen müsste man am besten über das User-Interface setzen könnnen.</p>
<i>[LAMBDA-149] created by lucas</i>

### PR DESCRIPTION
**Pull request by dragetd**
Fixes #143

We are trying to get the feeds accepted for some podcast indices. One example is iTunes, which after some time refuses to add LambdaCast feeds if they lack a category.

This PR add the ability to set a category. It also drops 'south' and uses the Django built-in migration for databases. We tested it not only on fresh but also old databases of a test-LambdaCast instance.

It would be great to have the category as part of the feeds and be able to list them on platforms like iTunes. Feedback and support is very much appreciated. Maybe you could even take a backup of theradio.cc database and deploy it there. Or if there is some way to help, let me know!

The code is almost completely authored by theradialactive <account@d2ns.org>, since my Python-skills are mediocre at best.

### CODE REVIEW

**Comment by dragetd** [comment]
@SuperTux88 Would love some feedback if you find some time. :) We could use it to add our podcast to some more indicies.

**Comment by dragetd** [comment]
@SuperTux88 The lack of categories is currently blocking our ability to get The LinuxLounge listed on some indexes.

Maybe we can discuss deploying this branch on theRadioCC without merging it yet into the official project? What do you think? :)

**Comment by dragetd** [comment]
@SuperTux88 Any news / perspective when this might be possible?

File: portal/models.py [hunk 1]
```diff
@@ -91,7 +82,7 @@ class MediaItem(models.Model):
     store your files '''
     title = models.CharField(_(u"Title"),max_length=200)
     slug = AutoSlugField(populate_from='title',unique=True,verbose_name=_(u"Slug"),help_text=_(u"Slugs are parts of an URL that you can define"))
-    date = models.DateField(_(u"Date"),help_text=_(u"Upload or record date"),default=lambda:localize_input(date.today()))
+    date = models.DateField(_(u"Date"),help_text=_(u"Upload or record date"),null=True)
```
Comment (SuperTux88) [inline portal/models.py:85]: Why was this changed? This should never be `null`. An item should always have a date.

**Review by SuperTux88 — CHANGES_REQUESTED** [review]
First, sorry this took so long.

The change of the complete migration system makes this change a lot more complex than just adding two fields. I don't exactly understand how the two systems differ and how easy it is to switch from one to the other. How is the update process? And does the installation documentation need to be updated or does this still work the same?

I also noticed the new migrations contain stuff like `help_text` or `verbose_name` which shouldn't have anything to do with migrations or the database, since that's only information for the UI, so this looks like code duplication that shouldn't be needed? :man_shrugging: 

Then I noticed that only the `0001_initial` migrations where changed to the new migration system, all the other migrations are still old with "south". So they probably need to be changed too? Does the new migration system know which old migrations were already done and which still need to be done? Or if the new migration system is a breaking change and can't keep the other migrations, then it needs to be documented that everything needs to be updated to the last version using still "south" before switching to the new migrations, and then all the old "south" migrations need to be deleted?

There is also no migration actually adding the new columns for the category. So updating an existing database that was created before this change actually fails with `Unknown column 'portal_channel.itunes_main_category' in 'field list'`.

File: portal/models.py (hunk 1 above)
Comment (TheRadialActive) [reply portal/models.py:85]: You're right. There should be a default. I removed it, because lambda threw an error. 
So the fix would be, @dragetd, to add a function called `localized_today` above the class MediaItem: 

```
def localized_today():
	return [localize_input(date.today())]
``

And then change the marked `null=True` to  `default=localized_today`.

Comment (TheRadialActive) [reply portal/models.py:85]: You're right. There should be a default. I removed it, because lambda threw an error: [ValueError: Cannot serialize function: lambda while using makemigrations](https://stackoverflow.com/questions/49642982/valueerror-cannot-serialize-function-lambda-while-using-makemigrations)

I've tried some things, but all the solutions had it's downsides.

**Comment by dragetd** [comment]
Please excuse the lack of answer from our side now. =/

Thank you very much for taking the time and looking over it. We will come back to this in our next meeting, currently I do not have any personal capacity to take this up. x.x

**Comment by dragetd** [comment]
Okay, we digged through the code. It was stupid to update the migrations away from south and try to add the category in the same MR. You are right, there is a lot of stuff in here that does not need to be changed.

Thank you very much for your review, but we would like to suggest a different approach!
Instead of hacking features into the old codebase, we would like to try to modernize the codebase first:

- We would want to remove some hard-to-update features like torrent support or tag-support first
- Remove the last call to the pytranscode / ffmpeg and replace it with a native ffmpeg call
- Migrate from South to Django Migrations
- Replace Django-Tasks with something modern, e.g. celery
- Port it to python 3 and a more current Django version.

And once we are there, then we can approach new features.

This makes this PR obsolete.

**Comment by SuperTux88** [comment]
Yes, that makes sense. Mixing big refactorings with features often is a bit problematic and it's definitely easier to split them.

> This makes this PR obsolete.

Do you want to close this and then do a new PR with the changes that are needed after the refactorings and modernize is done? Or do you want to keep this open and then just rebase on the refactorings later?

**Comment by dragetd** [comment]
Closing, sorry, forgot! :D
